lisa/src/main/java/it/unive/lisa/analysis/nonrelational/heap/NonRelationalHeapDomain.java

```diff
@@ -17,4 +19,46 @@
  */
 public interface NonRelationalHeapDomain<T extends NonRelationalHeapDomain<T>>
 		extends NonRelationalDomain<T, SymbolicExpression, HeapEnvironment<T>>, HeapSemanticOperation {
+
+	@Override
+	public default HeapEnvironment<T> assume(HeapEnvironment<T> environment, SymbolicExpression expression,
+			ProgramPoint pp) throws SemanticException {
+		return environment;
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public default T glb(T other) throws SemanticException {
```

Comment:
This should be defined only once in `NonRelationalDomain` instead of having a duplicate definition in `NonRelationalHeapDomain` and `NonRelationalValueDomain`. Also, no default definition should be present: as for other lattice operations, default behavior is to be implemented in a `BaseNonRelationalHeapDomain` instead of here.